pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -113,32 +113,11 @@ private static void validateSelectionClause(Map<Identifier, Expression> aliasMap
     // Sanity check on selection expression shouldn't use alias reference.
```

Comment:
We should not remove the sanity check on usage of alias reference
E.g. we should reject the following query:
```
SELECT colA AS a, foo(a, colB) FROM table
```

Another way is to replace the alias reference in selection as well

Reply:
Added it back, basically remove colA in aliasMap for case like `SELECT colA AS colA`.

Reply:
I think the correct fix should be just not adding it to the aliasMap?
You can rewrite the query before handling the alias by replacing `a AS a` to `a`

Reply:
done.